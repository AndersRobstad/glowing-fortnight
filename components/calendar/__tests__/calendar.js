import React from 'react';
import Calendar from '../calendar';
import calendarData from './calendarData.json';
import { shallow, mount } from 'enzyme';

const beforeDate = new Date('2017-08-01T00:00:00Z').valueOf();
const duringDate = new Date('2017-08-20T14:00:00Z').valueOf();
const afterDate = new Date('2017-09-01T00:00:00Z').valueOf();

it('renders correctly before fadderuker', () => {
  Date.now = jest.fn(() => beforeDate);
  const wrapper = shallow(<Calendar events={calendarData} error={null} />);
  expect(wrapper.getElements()).toMatchSnapshot();
});

it('renders correctly during fadderuker', () => {
  Date.now = jest.fn(() => duringDate);
  const wrapper = shallow(<Calendar events={calendarData} error={null} />);
  expect(wrapper.getElements()).toMatchSnapshot();
});

it('renders correctly after fadderuker', () => {
  Date.now = jest.fn(() => afterDate);
  const wrapper = shallow(<Calendar events={calendarData} error={null} />);
  expect(wrapper.getElements()).toMatchSnapshot();
});

it('renders past events after click', () => {
  Date.now = jest.fn(() => duringDate);
  const wrapper = shallow(<Calendar events={calendarData} error={null} />);
  wrapper.find('.cal-button--preDays').simulate('click');
  expect(wrapper.getElements()).toMatchSnapshot();
});

it('renders error', () => {
  const wrapper = shallow(<Calendar events={[]} error="Error message" />);
  expect(wrapper.getElements()).toMatchSnapshot();
});

it('renders loading message', () => {
  const wrapper = shallow(<Calendar events={[]} error={null} />);
  expect(wrapper.getElements()).toMatchSnapshot();
});

it('sets event to active after click', () => {
  Date.now = jest.fn(() => duringDate);
  const wrapper = mount(<Calendar events={calendarData} error={null} />);
  wrapper.find('.cal-button--preDays').simulate('click');
  let expectedIndex = 2;
  // TODO: This is likely not the best way to test this
  let eventHeader = wrapper
    .find('Event')
    .findWhere((n) => n.props().index == expectedIndex)
    .find('.cal-event-header');
  eventHeader.simulate('click');
  expect(wrapper.state('active')).toEqual(expectedIndex);

  expectedIndex = 0;
  eventHeader = wrapper
    .find('Event')
    .findWhere((n) => n.props().index == expectedIndex)
    .find('.cal-event-header');
  eventHeader.simulate('click');
  expect(wrapper.state('active')).toEqual(expectedIndex);
});
