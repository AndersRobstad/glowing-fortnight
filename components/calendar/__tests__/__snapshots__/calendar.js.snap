// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly after fadderuker 1`] = `
<div>
  <div
    className="cal-timeline"
  />
  
  <button
    className="cal-button--preDays"
    onClick={[Function]}
  >
    Vis tidligere arrangementer
  </button>
</div>
`;

exports[`renders correctly before fadderuker 1`] = `
<div>
  <div
    className="cal-timeline"
  />
  
  
  <div
    className="cal-day"
  >
    <h2
      className="cal-day--string"
    >
      Monday 14. August
    </h2>
    <div
      className="cal-event"
      id="event-0"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          15:00
        </p>
        <h3
          className="cal-event-title"
        >
          Ny-på-informatikk
        </h3>
      </div>
      <p
        className="cal-event-content cal-event-content-active"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Velkommen til Trondheim, informatikk og Online</p>
          <p>Lurer du på hvordan du skal komme deg rundt på Gløshaugen? Hvor de viktigste stedene i Trondheim er? Er det første gang du flytter hjemme fra og du lurer på hvordan i all verden du skal klare deg på bare lånet? Hva er det med de mattefagene alle snakker om? Og trenger man egentlig kjøpe pensum-bøker, det er jo så dyrt! Da er ny-på-informatikk-kurset noe for deg! En kjapp gjennomgang av det viktigste du trenger å vite som ny student på informatikk, hvordan du gjør deg klar til studiestart, fadderukene og alle mulighetene Trondheim har å by på.</p>
          <p>Sted: Kommer</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content cal-event-content-active cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        16:00
      </p>
    </div>
    <br />
  </div>
  <div
    className="cal-day"
  >
    <h2
      className="cal-day--string"
    >
      Sunday 20. August
    </h2>
    <div
      className="cal-event"
      id="event-1"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          14:00
        </p>
        <h3
          className="cal-event-title"
        >
          Skattejakt
        </h3>
      </div>
      <p
        className="cal-event-content "
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Nå er det på tide å finne frem sin indre skatteplyndrer, for i dag er det skattejakt! Møt opp til anvist tid utenfor P15.</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content  cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        17:00
      </p>
    </div>
    <br />
  </div>
  <div
    className="cal-day"
  >
    <h2
      className="cal-day--string"
    >
      Monday 21. August
    </h2>
    <div
      className="cal-event"
      id="event-2"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          17:00
        </p>
        <h3
          className="cal-event-title"
        >
          Hybel til Hybel
        </h3>
      </div>
      <p
        className="cal-event-content "
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Blåmandag er et kjent fenomen for de fleste. Denne mandagen blir alt annet enn blå, for i dag er det duket for Hybel-Hybel!</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content  cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        23:00
      </p>
    </div>
    <div
      className="cal-event"
      id="event-3"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          23:00
        </p>
        <h3
          className="cal-event-title"
        >
          Etterfest
        </h3>
      </div>
      <p
        className="cal-event-content "
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Etter konkurranserunden, er det etterfest på Cafè 3b.</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content  cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        23:59
      </p>
    </div>
    <br />
  </div>
  <div
    className="cal-day"
  >
    <h2
      className="cal-day--string"
    >
      Tuesday 22. August
    </h2>
    <div
      className="cal-event"
      id="event-4"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          17:00
        </p>
        <h3
          className="cal-event-title"
        >
          Facebook-Fest
        </h3>
      </div>
      <p
        className="cal-event-content "
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>I dag er det duket for den legendariske Facebook-Festen! Faddergruppene står for faddervors før alle møtes på Nattergalen.</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content  cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        23:59
      </p>
    </div>
    <div
      className="cal-event"
      id="event-5"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          18:00
        </p>
        <h3
          className="cal-event-title"
        >
          Kino og Middag
        </h3>
      </div>
      <p
        className="cal-event-content "
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Alternativet til kveldens fest er en tur på kino, etterfulgt av felles middag. Perfekt mulighet for å bli kjent med medstudentene dine.</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content  cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        23:30
      </p>
    </div>
    <br />
  </div>
  <div
    className="cal-day"
  >
    <h2
      className="cal-day--string"
    >
      Wednesday 23. August
    </h2>
    <div
      className="cal-event"
      id="event-6"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          14:00
        </p>
        <h3
          className="cal-event-title"
        >
          Grilling
        </h3>
      </div>
      <p
        className="cal-event-content "
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>I dag tar vi det rolig, og griller i høyskoleparken. Gratis mat = Studentens beste venn.</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content  cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        17:00
      </p>
    </div>
    <br />
  </div>
</div>
`;

exports[`renders correctly during fadderuker 1`] = `
<div>
  <div
    className="cal-timeline"
  />
  
  <button
    className="cal-button--preDays"
    onClick={[Function]}
  >
    Vis tidligere arrangementer
  </button>
  <div
    className="cal-day"
  >
    <h2
      className="cal-day--string"
    >
      Sunday 20. August
    </h2>
    <div
      className="cal-event"
      id="event-1"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          14:00
        </p>
        <h3
          className="cal-event-title"
        >
          Skattejakt
        </h3>
      </div>
      <p
        className="cal-event-content cal-event-content-active"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Nå er det på tide å finne frem sin indre skatteplyndrer, for i dag er det skattejakt! Møt opp til anvist tid utenfor P15.</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content cal-event-content-active cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        17:00
      </p>
    </div>
    <br />
  </div>
  <div
    className="cal-day"
  >
    <h2
      className="cal-day--string"
    >
      Monday 21. August
    </h2>
    <div
      className="cal-event"
      id="event-2"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          17:00
        </p>
        <h3
          className="cal-event-title"
        >
          Hybel til Hybel
        </h3>
      </div>
      <p
        className="cal-event-content "
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Blåmandag er et kjent fenomen for de fleste. Denne mandagen blir alt annet enn blå, for i dag er det duket for Hybel-Hybel!</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content  cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        23:00
      </p>
    </div>
    <div
      className="cal-event"
      id="event-3"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          23:00
        </p>
        <h3
          className="cal-event-title"
        >
          Etterfest
        </h3>
      </div>
      <p
        className="cal-event-content "
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Etter konkurranserunden, er det etterfest på Cafè 3b.</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content  cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        23:59
      </p>
    </div>
    <br />
  </div>
  <div
    className="cal-day"
  >
    <h2
      className="cal-day--string"
    >
      Tuesday 22. August
    </h2>
    <div
      className="cal-event"
      id="event-4"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          17:00
        </p>
        <h3
          className="cal-event-title"
        >
          Facebook-Fest
        </h3>
      </div>
      <p
        className="cal-event-content "
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>I dag er det duket for den legendariske Facebook-Festen! Faddergruppene står for faddervors før alle møtes på Nattergalen.</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content  cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        23:59
      </p>
    </div>
    <div
      className="cal-event"
      id="event-5"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          18:00
        </p>
        <h3
          className="cal-event-title"
        >
          Kino og Middag
        </h3>
      </div>
      <p
        className="cal-event-content "
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Alternativet til kveldens fest er en tur på kino, etterfulgt av felles middag. Perfekt mulighet for å bli kjent med medstudentene dine.</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content  cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        23:30
      </p>
    </div>
    <br />
  </div>
  <div
    className="cal-day"
  >
    <h2
      className="cal-day--string"
    >
      Wednesday 23. August
    </h2>
    <div
      className="cal-event"
      id="event-6"
    >
      <div
        className="cal-event-indicator"
      />
      <div
        className="cal-event-header"
        onClick={[Function]}
      >
        <p
          className="cal-event-date"
        >
          14:00
        </p>
        <h3
          className="cal-event-title"
        >
          Grilling
        </h3>
      </div>
      <p
        className="cal-event-content "
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>I dag tar vi det rolig, og griller i høyskoleparken. Gratis mat = Studentens beste venn.</p>
          ",
          }
        }
      />
      <p
        className="cal-event-content  cal-event-endtime"
      >
        Antatt sluttidspunkt: 
        17:00
      </p>
    </div>
    <br />
  </div>
</div>
`;

exports[`renders error 1`] = `
<p
  className="component"
>
  En uventet feil har oppstått ved henting av program. Vennligst prøv igjen senere.
</p>
`;

exports[`renders loading message 1`] = `
<h2
  className="component"
>
  Laster inn kalender
</h2>
`;
